### PR TITLE
Remove support for old when rules

### PR DESCRIPTION
**Pull request by ONS-Guilhem-Forey**
### PR Context
We want to get rid of the deprecated WHEN rules.
They include:
- Routing conditions
- Skip conditions
- Content variants
- Question variants

### How to review
Ticket is made of two PRs, validator and runner.

I suggest reviewing the validator first.

### Checklist

- [x] Upgrade all schemas to the new WHEN rules
- [x] Remove code related to deprecated when rules
- [x] Remove deprecated tests

### CODE REVIEW

**Comment by ONS-Guilhem-Forey** [comment]
[Remove support for old when rules (3 SP)](https://trello.com/c/uCidXQd2/5295-remove-support-for-old-when-rules-3-sp)

File: app/validators/routing/new_routing_validator.py
Comment (berroar) [inline app/validators/routing/new_routing_validator.py:1]: Could this file have been renamed and the old one removed rather than updating the old with changes? As was done in other places in this PR

Comment (ONS-Guilhem-Forey) [reply app/validators/routing/new_routing_validator.py:1]: As explained in Slack - file was renamed and old one remove, but Git diff doesn't understand it as such

**Comment by MebinAbraham** [comment]
Temporary do not merge label. Can be removed once runner PR is ready to merge.

**Review by berroar — APPROVED** [review]

File: app/validators/sections/section_validator.py
Comment (MebinAbraham) [inline app/validators/sections/section_validator.py:49]: Is this needed?. It should now always be a dict?.

File: app/validators/sections/section_validator.py
Comment (MebinAbraham) [inline app/validators/sections/section_validator.py:106]: Is the isinstance check needed anymore?.

File: app/validators/sections/section_validator.py
Comment (MebinAbraham) [inline app/validators/sections/section_validator.py:156]: Same as above. Should be a dict?. The above lines default for `.get()` needs to change.

**Review by MebinAbraham — COMMENTED** [review]
This can be removed: `test_question_variants_new_when_rule.json`

File: app/validators/sections/section_validator.py
Comment (ONS-Guilhem-Forey) [reply app/validators/sections/section_validator.py]: Now only testing if `section_enabled` is Falsy

File: app/validators/sections/section_validator.py
Comment (ONS-Guilhem-Forey) [reply app/validators/sections/section_validator.py]: Now only doing:

```python
if "skip_conditions" in schema_element:
```

File: app/validators/sections/section_validator.py
Comment (ONS-Guilhem-Forey) [reply app/validators/sections/section_validator.py]: Replaced with 

```python
if when_clause := variant.get("when", None):
```

File: app/validators/sections/section_validator.py
Comment (MebinAbraham) [inline app/validators/sections/section_validator.py:104]: Minor could make use of walrus operator.

File: app/validators/sections/section_validator.py
Comment (MebinAbraham) [inline app/validators/sections/section_validator.py:153]: `.get()` doesn't need `None`, since that is the default anyway.

File: tests/schemas/valid/test_section_enabled.json
Comment (MebinAbraham) [inline tests/schemas/valid/test_section_enabled.json:30]: Should use `Hub` as although not enforced yet, multiple section should be hub.

**Review by MebinAbraham — COMMENTED** [review]
These methods in `questionnaire_schema.py seem unused now, if so should be removed:

- has_default_route
- get_routing_when_list
- is_contained_in_dict_list

File: app/validators/sections/section_validator.py
Comment (ONS-Guilhem-Forey) [reply app/validators/sections/section_validator.py]: Made use of the walrus

File: app/validators/sections/section_validator.py
Comment (ONS-Guilhem-Forey) [reply app/validators/sections/section_validator.py]: Addressed

File: tests/schemas/valid/test_section_enabled.json
Comment (ONS-Guilhem-Forey) [reply tests/schemas/valid/test_section_enabled.json]: Changed to Hub

**Review by MebinAbraham — APPROVED** [review]
👍

**Pull request merged**
